Makefile

```diff
@@ -40,17 +40,29 @@ __toolversion = $(shell $(GIT_ROOT)/scripts/toolversion $(1))
 # Main targets
 
 clean: ##@prepare Remove all output folders
-	git clean -dxf -f -e android/local.properties
+	git clean -dxf -f
 
 setup: ##@prepare Install all the requirements for status-react
 	@./scripts/setup
-	@./scripts/run-environment-check.sh setup
+
+shell: ##@prepare Enter into a preconfigured shell
+ifndef IN_NIX_SHELL
+	@echo "Configuring Nix shell..."
+	@nix-shell
+else
+	@echo "Nix shell is already active"
+endif
 
 prepare-desktop: ##@prepare Install desktop platform dependencies and prepare workspace
+ifndef IN_NIX_SHELL
```

Comment:
You can make a target called `nix-shell-check` that would have this `ifndef` and re-use it in other targets.

Reply:
But that won't prevent the invoked target (`prepare-desktop` in this case) from running to completion, right?

Reply:
It will if you add `exit 1`. Or use `error` like I do here:
https://github.com/status-im/status-go/blob/72906ac6553869f6ac6df4e294a91c184cb320a4/_assets/compose/mailserver/Makefile#L23-L25